tip-distributor/src/claim_mev_workflow.rs

```diff
@@ -32,121 +39,343 @@ pub enum ClaimMevError {
-            let min_rent_per_claim = rpc_client.get_minimum_balance_for_rent_exemption(ClaimStatus::SIZE).await.expect("Failed to calculate min rent");
-            let desired_balance = (node_count as u64).checked_mul(min_rent_per_claim.checked_add(DEFAULT_TARGET_LAMPORTS_PER_SIGNATURE).unwrap()).unwrap();
-            if start_balance < desired_balance {
-                let sol_to_deposit = desired_balance.checked_sub(start_balance).unwrap().checked_add(LAMPORTS_PER_SOL).unwrap().checked_sub(1).unwrap().checked_div(LAMPORTS_PER_SOL).unwrap(); // rounds up to nearest sol
-                panic!("Expected to have at least {} lamports in {}, current balance is {} lamports, deposit {} SOL to continue.",
-                       desired_balance, &keypair.pubkey(), start_balance, sol_to_deposit)
+    let tree_nodes = merkle_trees
+        .generated_merkle_trees
+        .iter()
+        .flat_map(|tree| &tree.tree_nodes)
+        .collect_vec();
+    let tree_node_count = tree_nodes.len();
+    let stake_acct_min_rent = rpc_client
+        .get_minimum_balance_for_rent_exemption(StakeStateV2::size_of())
+        .await
+        .expect("Failed to calculate min rent");
+    let mut skipped_merkle_root_count: usize = 0;
+    let mut zero_lamports_count: usize = 0;
+    let mut already_claimed_count: usize = 0;
+    let mut below_min_rent_count: usize = 0;
+    let mut instructions =
+        Vec::with_capacity(tree_nodes.iter().filter(|node| node.amount > 0).count());
+
+    let account_fetch_start = Instant::now();
+    let tdas = get_batched_accounts(
+        &rpc_client,
+        merkle_trees
+            .generated_merkle_trees
+            .iter()
+            .map(|tree| tree.tip_distribution_account)
+            .collect_vec(),
+    )
+    .await
+    .unwrap()
+    .into_iter()
+    .filter_map(|(pubkey, maybe_account)| {
+        let account = match maybe_account {
+            Some(a) => a,
+            None => {
+                datapoint_warn!(
+                    "claim_mev_workflow-account_error",
+                    ("pubkey", pubkey.to_string(), String),
+                    ("pubkey_type", "tip_distribution_account", String),
+                    ("error", 1, i64),
+                    ("err_type", "fetch", String),
+                    ("err_str", "Failed to fetch Account", String)
+                );
+                return None;
+            }
+        };
+
+        let account = match TipDistributionAccount::try_deserialize(&mut account.data.as_slice()) {
+            Ok(a) => a,
+            Err(e) => {
+                datapoint_warn!(
+                    "claim_mev_workflow-account_error",
+                    ("pubkey", pubkey.to_string(), String),
+                    ("pubkey_type", "tip_distribution_account", String),
+                    ("error", 1, i64),
+                    ("err_type", "deserialize_tip_distribution_account", String),
```

Comment:
seems redundant, how about just changing "error" to be "deser_error_count"?

Reply:
the rationale for "error", 1 is mainly for the charts, and "err_type" for debugging